CHANGELOG.md

```diff
@@ -605,6 +605,18 @@ Other minor additions
   map-∷=     : map f (xs [ k ]∷= v) ≡ map f xs [ cast eq k ]∷= f v
   length-─   : length (xs ─ k) ≡ pred (length xs)
   map-─      : map f (xs ─ k) ≡ map f xs ─ cast eq k
+
+  length-applyUpTo     : length (applyUpTo     f n) ≡ n
+  length-applyDownFrom : length (applyDownFrom f n) ≡ n
+  length-upTo          : length (upTo            n) ≡ n
+  length-downFrom      : length (downFrom        n) ≡ n
+
+  applyUpTo-lookup     : lookup (applyUpTo     f n) i ≡ f (toℕ i)
+  applyDownFrom-lookup : lookup (applyDownFrom f n) i ≡ f (n ∸ (suc (toℕ i)))
```

Comment:
This doesn't seem to follow the same convention as `length-applyUpTo`.

Reply:
Sorry I must be being slow. Why would `applyDownFrom-lookup` follow the same convention as `length-applyUpTo`?

Reply:
I meant the naming didn't follow the same convention. I think it's easier if we
always name lemmas of the form `f (g x) y == z` something like `f-g` rather
than sometimes `f-g` and sometimes `g-f`.

Reply:
Haha, I've never actually realised that convention existed. Now you point it out to me though, it's obvious! I've always wondered why `length-OP` wasn't `OP-length'...

I've changed the names.